src/abstractdataframe/abstractdataframe.jl

```diff
@@ -970,15 +970,47 @@ julia> dropmissing(df, [:x, :y])
 @inline function dropmissing(df::AbstractDataFrame,
                              cols::Union{ColumnIndex, MultiColumnIndex}=:;
                              view::Bool=false, disallowmissing::Bool=!view)
+    # Identify Bool mask of which rows have no Missings
     rowidxs = completecases(df, cols)
     if view
         if disallowmissing
             throw(ArgumentError("disallowmissing=true is incompatible with view=true"))
         end
         return Base.view(df, rowidxs, :)
     else
-        newdf = df[rowidxs, :]
-        disallowmissing && disallowmissing!(newdf, cols)
+        # Faster when there are many columns (Indexing with Integers than via Bool mask)
+        # or when there are many Missing (as we skip a lot of iterations)
+        selected_rows = _findall(rowidxs)
+        new_columns = Vector{AbstractVector}(undef, ncol(df))
+        df_columns = df isa DataFrame ? _columns(df) : collect(AbstractVector, eachcol(df))
+
+        # What column indices should disallowmissing be applied to
+        cols_inds = index(df)[cols]
+
+        # Threading decision rule borrowed from `_threaded_getindex`
+        if Threads.nthreads() > 1 && ncol(df) > 1 && length(selected_rows) >= 1_000_000
+            @sync for i in eachindex(new_columns)
+                # for each column, check if disallowmissing should be applied
+                Threads.@spawn if disallowmissing && (i in cols_inds)
+                    # implicit imports of functions due to name clash with keyword arguments!
```

Comment:
"Implicit" should be "explicit". But better drop the comment as this is a common pattern.
```suggestion
```